pkg/console/telemetry/telemetry.go

```diff
@@ -0,0 +1,157 @@
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	v1 "k8s.io/client-go/listers/core/v1"
+
+	"github.com/openshift/console-operator/pkg/api"
+	deploymentsub "github.com/openshift/console-operator/pkg/console/subresource/deployment"
+)
+
+const (
+	TelemeterClientDeploymentName      = "telemeter-client"
+	TelemetryAnnotationPrefix          = "telemetry.console.openshift.io/"
+	TelemeterClientDeploymentNamespace = "openshift-monitoring"
+	PullSecretName                     = "pull-secret"
+)
+
+func IsTelemeterClientAvailable(deploymentLister appsv1listers.DeploymentLister) (bool, error) {
+	deployment, err := deploymentLister.Deployments(TelemeterClientDeploymentNamespace).Get(TelemeterClientDeploymentName)
+
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	return deploymentsub.IsAvailable(deployment), nil
+}
+
+func GetClusterID(clusterVersionLister configlistersv1.ClusterVersionLister) (string, error) {
+	cv, cvErr := clusterVersionLister.Get(api.VersionResourceName)
+	if cvErr != nil {
+		return "", cvErr
+	}
+	return string(cv.Spec.ClusterID), nil
+}
+
+type DockerConfig struct {
+	Auths map[string]DockerAuthEntry `json:"auths"`
+}
+
+// DockerAuthEntry contains the base64-encoded auth credentials for a Docker registry.
+type DockerAuthEntry struct {
+	Auth string `json:"auth"`
+}
+
+func GetAccessToken(secretsLister v1.SecretLister) (string, error) {
+	secret, err := secretsLister.Secrets(api.OpenShiftConfigNamespace).Get(PullSecretName)
+	if err != nil {
+		return "", err
+	}
+
+	configBytes, ok := secret.Data[".dockerconfigjson"]
+	if !ok {
+		return "", fmt.Errorf("failed to parse .dockerconfigjson field from pull-secret")
+	}
+	var config DockerConfig
+	err = json.Unmarshal(configBytes, &config)
+	if err != nil {
+		fmt.Println("Error decoding JSON:", err)
+		return "", err
+	}
+	authsBytes, ok := config.Auths["cloud.openshift.com"]
+	if !ok {
+		return "", fmt.Errorf("failed to parse 'cloud.openshift.com' field from pull-secret")
+	}
+	return authsBytes.Auth, nil
+}
+
+// Needed to create our own types for OCM Subscriptions since their types and client are useless
+// https://github.com/openshift-online/ocm-sdk-go/blob/main/accountsmgmt/v1/subscription_client.go - everything private
+// https://github.com/openshift-online/ocm-sdk-go/blob/main/accountsmgmt/v1/subscriptions_client.go#L38-L41 - useless client
+type OCMAPIResponse struct {
+	Items []Organization `json:"items"`
+}
+type Organization struct {
+	OrganizationID string `json:"organization_id"`
+}
+
+// GetOrganizationID fetches the organization ID using the cluster ID and access token
+func GetOrganizationID(clusterID, accessToken string) (string, error) {
+	u, err := buildURL(clusterID)
+	if err != nil {
+		return "", err // more contextual error handling can be added here if needed
+	}
+
+	req, err := createRequest(u, clusterID, accessToken)
+	if err != nil {
+		return "", err
+	}
+
+	client := &http.Client{}
```

Comment:
Should we use the http proxy configuration here to ensure that it's working on airgapped clusters?

```suggestion
	client := &http.Client{
		Timeout: 30 * time.Second,
		Transport: &http.Transport{
			Proxy: http.ProxyFromEnvironment,
		},
	}
```